src/Java.Interop.Tools.Generator/Metadata/NamespaceTransform.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Xml.Linq;
+using Xamarin.Android.Tools;
+
+namespace Java.Interop.Tools.Generator
+{
+	public class NamespaceTransform
+	{
+		public string OldValue { get; }
+		public string NewValue { get; }
+		public bool IsStartsWith { get; }
+		public bool IsEndsWith { get; }
+
+		public NamespaceTransform (string oldValue, string newValue)
+		{
+			OldValue = oldValue;
+			NewValue = newValue;
+
+			if (OldValue.EndsWith (".", StringComparison.Ordinal)) {
```

Comment:
What should happen if `OldValue` is `".Foo."`?  This isn't explicitly called out in the PR message, but *a* meaning can be inferred: it would only match when `OldValue` is `Foo`, case-insensitive.  It wouldn't match `Example.Foo` or `Foo.Example`.  (Why would you want this?  I don't know, but this behavior is implied in the PR message.)

…and that kinda works *here*, in the `NamespaceTransform` constructor: `IsStartsWith=true` would be set, and `IsEndsWith=true` would be set…